llvm/test/CodeGen/AArch64/O0-pipeline.ll

```diff
@@ -31,6 +31,10 @@
 ; CHECK-NEXT:       AArch64 Stack Tagging
 ; CHECK-NEXT:       SME ABI Pass
 ; CHECK-NEXT:       Exception handling preparation
+; CHECK-NEXT:       Dominator Tree Construction
+; CHECK-NEXT:       Basic Alias Analysis (stateless AA impl)
+; CHECK-NEXT:       Function Alias Analysis Results
```

Comment:
Presumably the extra analysis passes are the reason for the compile-time regressions. While you may be skipping the execution of the pass itself, you are not skipping the execution of the analyses.

Reply:
this is where having the new pass manager for codegen would be nice

Reply:
I suspected this might be an issue. Is it possible to "lazily" request analysis results based on some logic in the pass itself? If not, maybe we can't enable this pass by default after all.